mu/logic.py

```diff
@@ -1074,28 +1096,51 @@ def show_admin(self, event=None):
 
         Ensure any changes to the envars is updated.
         """
-        logger.info('Showing logs from {}'.format(LOG_FILE))
+        logger.info('Showing admin with logs from {}'.format(LOG_FILE))
         envars = '\n'.join(['{}={}'.format(name, value) for name, value in
                             self.envars])
         settings = {
             'envars': envars,
             'minify': self.minify,
             'microbit_runtime': self.microbit_runtime,
         }
+        packages = installed_packages()
         with open(LOG_FILE, 'r', encoding='utf8') as logfile:
-            new_settings = self._view.show_admin(logfile.read(), settings)
-            self.envars = extract_envars(new_settings['envars'])
-            self.minify = new_settings['minify']
-            runtime = new_settings['microbit_runtime'].strip()
-            if runtime and not os.path.isfile(runtime):
-                self.microbit_runtime = ''
-                message = _('Could not find MicroPython runtime.')
-                information = _("The micro:bit runtime you specified ('{}') "
-                                "does not exist. "
-                                "Please try again.").format(runtime)
-                self._view.show_message(message, information)
+            new_settings = self._view.show_admin(logfile.read(), settings,
+                                                 '\n'.join(packages))
```

Comment:
Don't think this is really a problem, but everything below this line could be unindented out of the `with` statement, so that the log file handler can be closed.
Line 1129 at the end of this function (inside this `with` statement) does also write to the log, but it seems to work fine.

Reply:
You're right! Good spot. :-)